opencti-platform/opencti-graphql/src/manager/taskManager.js

```diff
@@ -406,7 +407,7 @@ const executeShare = async (context, user, actionContext, element) => {
   for (let indexCreate = 0; indexCreate < values.length; indexCreate += 1) {
     const target = values[indexCreate];
     const currentGrants = element[buildRefRelationKey(RELATION_GRANTED_TO)] ?? [];
-    if (!currentGrants.includes(target)) {
+    if (!currentGrants.includes(target) && objectOrganization.isRefExistingForTypes(element.entity_type, ENTITY_TYPE_IDENTITY_ORGANIZATION)) {
```

Comment:
Can you explain the change?

Reply:
Had the answer in the description oft he PR

Reply:
Nop in fact even with description I do not understand how this line check that the relation GRANTED REF is possible between the 2 elements as we are not specifying the type of the ref anywhere ? 😅

Reply:
when we share an element with an `Organization`, we create the `granted` relationship if the element's granted relationships don't include the target (= if the element isn't already shared with this orga) and if it's possible to create a `granted` relationship for this entity. 
The method `isRefExistingForTypes(element, orga)` checks that the element is not an `Event`, `Organization`, `Sector` or `Location`.

Before the modification, this check was not done and the backend sent an error, as it was not possible to create a `granted` relationship between an `Organization` and a `Sector` or between 2 Organizations. (you can see the error in the ticket).

all unrestricted entities are listed in `stixDomainObject.ts` file line 182.
```
export const STIX_ORGANIZATIONS_UNRESTRICTED = [
  ABSTRACT_INTERNAL_OBJECT,
  ABSTRACT_INTERNAL_RELATIONSHIP,
  ABSTRACT_STIX_META_OBJECT,
  ABSTRACT_STIX_REF_RELATIONSHIP,
  ENTITY_TYPE_IDENTITY_ORGANIZATION,
  ENTITY_TYPE_IDENTITY_SECTOR,
  ENTITY_TYPE_LOCATION,
  ENTITY_TYPE_WORK, // Work is defined as an history object
  ENTITY_TYPE_TAXII_COLLECTION, // TODO TaxiiCollection must be migrate to add according parent types
  ENTITY_TYPE_INTERNAL_FILE, // TODO InternalFile must be migrate to add according parent types
];
```

Reply:
Thanks for the explanation.

So the function called `isRefExistingForTypes` checks if it is possible to have a `granted` ref between two types? That's quite confusing to me 😅

Reply:
yes this method is called when creating the relation and if it returns false, we throw an exception. The idea here was to call the method before trying to create a granted relationship that won't work and avoid unecessary errors in the logs.
![image](https://github.com/user-attachments/assets/dddfd2bf-a3c5-48fb-87d1-ff4d8a003fb5)